File: array/Queue_Dynamic.cs
using System;

public class ArrayMain
{
  public static void Main()
  {
    Queue queue = new Queue();

    queue.Enqueue(1);
    queue.Enqueue(2);
    queue.Enqueue(3);
    queue.Enqueue(4);
    queue.Enqueue(5);

    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
    Console.WriteLine("Dequeue > {0}", queue.Dequeue());
  }
}

public class Queue
{
  int[] queue;

  public Queue()
  {
    /* Inicializando a fila vazia. */
    queue = new int[0];
  }

  /* Altere o tamanho da fila para adicionar o elemento */
  public void Enqueue(int element)
  {
    Array.Resize(ref queue, queue.Length + 1);
    queue[queue.Length - 1] = element;
  }

  /* Guarde o elemento e altere o tamanho da fila */
  public int Dequeue()
  {
    int element = queue[queue.Length - 1];
    Array.Resize(ref queue, queue.Length - 1);

    return element;
  }
}

File: array/Queue_Dynamic.ex
defmodule Queue do
  def initialize, do: []
  
  def enqueue(queue, element), do:  queue ++ [element]

  def dequeue(queue) do
    {n, queue} = List.pop_at(queue, 0)

    {:ok, queue, n}
  end
end

queue = Queue.initialize

queue = Queue.enqueue(queue, 1)
queue = Queue.enqueue(queue, 2)
queue = Queue.enqueue(queue, 3)
queue = Queue.enqueue(queue, 4)
queue = Queue.enqueue(queue, 5)
# queue = Queue.enqueue(queue, 6)

{:ok, queue, n1} = Queue.dequeue(queue)
{:ok, queue, n2} = Queue.dequeue(queue)
{:ok, queue, n3} = Queue.dequeue(queue)
{:ok, queue, n4} = Queue.dequeue(queue)
{:ok, queue, n5} = Queue.dequeue(queue)
# {:error, queue, nil} = Queue.dequeue(queue)

IO.inspect queue
